azure/services/scalesets/scalesets.go

```diff
@@ -276,6 +278,18 @@ func (s *Service) patchVMSSIfNeeded(ctx context.Context, infraVMSS *azure.VMSS)
 		return nil, errors.Wrap(err, "failed to calculate maxSurge")
 	}
 
+	replicaseManagedByAutoscaler := s.Scope.HasReplicasExternallyManaged(ctx)
```

Comment:
I'm not sure how else we would be able to differentiate the reconcile loop based on the existence of the external autoscaler annotation. Open to suggestions.

Reply:
I just meant using `s.Scope.HasReplicasExternallyManaged(ctx)` directly instead of assigning it to a variable since it's only used once... not a huge deal though, definitely a nit